packages/maskbook/src/web3/suspends/getERC721TokenDetailed.ts

```diff
@@ -0,0 +1,38 @@
+import type { ERC721 } from '../../contracts/ERC721'
+import { formatChecksumAddress } from '../../plugins/Wallet/formatter'
+import { useERC721TokenContract } from '../contracts/useERC721TokenContract'
+import { useChainId } from '../hooks/useChainState'
+import { useSingleContractMultipleData } from '../hooks/useMulticall'
+import { ERC721TokenDetailed, EthereumTokenType } from '../types'
+
+const cache = new Map<string, ERC721TokenDetailed | undefined>()
+async function ERC721TokenDetailed_(address: string, token?: Partial<ERC721TokenDetailed>) {
+    const chainId = useChainId()
+    const erc721TokenContract = useERC721TokenContract(address)
```

Comment:
IMO this is not correct. You're violating React hooks rules. Because this function is conditionally called. If you're intended to do this, please call those hooks in `getERC721TokenDetailed` unconditionally and rename it to `useERC721TokenDetailed`.